app/javascript/packs/homePage.jsx

```diff
@@ -62,9 +62,13 @@ function trackTagCogIconClicks() {
     });
 }
 
+function removeLocalePath(pathname) {
+  return pathname.replace(/^\/locale\/[a-zA-Z-]+\/?/, '/');
```

Comment:
The issue here is primarily around line 75, where we're comparing the `pathname` to some fixed paths that have some baked-in assumptions about how paths are formed (assumptions which, as it turns out, ignore `locale`).

So, this _works_, and fixes the immediate problem, but in the long term, we probably ought to refactor this towards having Rails indicate whether or not the current view _should_ render the sidebar — so the front-end can best handle `screen.width` and `sidebarContainer`, but the server-side can better handle the `pathname` bits.

But that can be later, better now to get this working. 

For `locale`, the `a-z` assumption fits the locales we have today, but another option here might be:

`/^\/locale\/([^\/]+\/?)/`

That is: the regex can treat the `locale` part of the path as a sequence of one or more non-slash characters, possibly followed by a slash. That would allow, assuming we don't otherwise refactor this by then, some future where locale might include region as well as language? (eg, `en-US`, `nl-NL`, etc.)

Reply:
Thank you for reviewing it! ❤️ The regex I wrote already supports `a-z`, `A-Z` and `-` because it has `-` in the `[]`.
I’m not sure what characters the `locale` can contain, but as far as I've checked [rails-i18n](https://github.com/svenfuchs/rails-i18n), perhaps this regexp is sufficient to support all of locales.

```js
'/locale/en-US/foo/bar'.replace(/^\/locale\/[a-zA-Z-]+\/?/, '/') // => '/foo/bar'
'/locale/nl-NL/foo/bar'.replace(/^\/locale\/[a-zA-Z-]+\/?/, '/') // => '/foo/bar'
```

Reply:
You're right — I missed that trailing `-`! I think we're good to go here 🎉